feat: Add custom notifications for each site

coolusaHD opened the pull request:
As described in https://github.com/orgs/upptime/discussions/930 , it would be nice to set diffrent notification for each site.

I know that the isse proposed to add that feature to send emails but for the first step this PR only adds webhooks.

Before I invest more time in that, it would be nice to know if my intensions are right.

@AnandChowdhary Do you have any ideas or comments on that ?

Currently I would change the current behaviour to only send general notifications for sites that have no custom notification type set and the custom ones for that they are configurrf

Furthermore I can imagine to refactor the `update` and `sendNotification` to make it more readable and easier to add more feature in the future.

Feel free to comment ✌️ 

Best @coolusaHD

coolusaHD commented:
@AnandChowdhary Any thoughts on this ?

AnandChowdhary commented:
Looks good to me! If I understand it correctly, this PR only adds support for custom webhooks right?

coolusaHD commented:
Currently yes but I am planning to add more.

coolusaHD commented:
@AnandChowdhary This PR is almost finished, before I change the docs I want to explain what I've done and ask for review / opinion ✌️ 

I refactored the sending functionality and extract each send function in his own file where the checks and the setup are also placed.
I added the rest of the notification methods to be configurable as `customNotification`.
Th only problem what I encountered was the override of the sms or email provider but I think it's a rare edge case when someone wants to send their sms or mail via a different provider.

Furthermore I implemented the custom action logger

coolusaHD commented:
@AnandChowdhary Any update on this ?
